code/src/mq_sensors_gas_curve_c.cpp

```diff
@@ -0,0 +1,44 @@
+#include "mq_sensors_gas_curve_c.hpp"
+#include "gas_enums.hpp"
+#include <array>
+
+namespace r2d2::gas_detection {
+    mq_sensors_gas_curve_c::mq_sensors_gas_curve_c(
+        const int &gas_id, const std::array<float, 3> &gas_curve)
+        : gas_id(gas_id), gas_curve(gas_curve) {
+    }
+
+    mq_sensors_gas_curve_c::mq_sensors_gas_curve_c(const int &gas_id)
+        : gas_id(gas_id) {
+        if (gas_id == static_cast<int>(r2d2::gas_detection::gas_type::LPG)) {
```

Comment:
A switch-case would be neater here, as you don't need the casts. The compiler will also warn you if you miss a gas type.

Reply:
Im currently working on solving this pr. Im using a switch case, but before the the switch case I still have to use 1 cast. Is this right or is there a way to do it without casting at all?